### PR TITLE
Tt 3384 feature/expose entity

### PR DESCRIPTION
**Pull request by sredxny**
## Description
Some IDPs just reject users to set as EntityID an URL value, according to the SAML Standard this value can be either a string or an URL, currently we set as EntityID in the SP the /metadata url and do not allow users to set a different value than this. Now, we expose that field and if not set then it will default to the metadata URL

## Related Issue
https://tyktech.atlassian.net/browse/TT-3384

## Motivation and Context
Be more standard compliant

## How This Has Been Tested
- Created profile as:
```
{
        "ID": "saml-sso-dash-login",
        "OrgID": "5e2091c4d4aefce60c04fb92",
        "ActionType": "GenerateOrLoginUserProfile",
        "Type": "redirect",
        "ProviderName": "SAMLProvider",
        "ProviderConfig": {
            "CertLocation": "/my-cert-location/cert.pem",
            "FailureRedirect": "http://tyk-dashboard:3000/?fail=true",
            "ForceAuthentication": true,
            "IDPMetaDataURL": "https://my-idp/federationmetadata.xml",
            "SAMLBaseURL": "http://localhost:3010/",
            "EntityId": ""
        },
        "IdentityHandlerConfig": {
            "DashboardCredential": "my-dash-creds"
        },
        "ReturnURL": "http://tyk-dashboard:3000/tap",
        "SSOOnlyForRegisteredUsers": false
    },
```
- In the IDP set as Entity ID = the metadata url. Performed the authentication flow, and it works
- In the IDP set as entity ID a differnt value, performed the authentication flow and it fails due Audience restriction (because those values should match)
- In my SP set a value for entity ID as "my-entity-id" and set the same value for EntityId in my IDP, performed the flow and it works because the values matches.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If pulling from your own
      fork, don't request your `master`!
- [x] Make sure you are making a pull request against the **`master` branch** (left side). Also, you should start
      *your branch* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
  - [ ] If new config option added, ensure that it can be set via ENV variable
- [ ] I have updated the documentation accordingly.
- [ ] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [ ] When updating library version must provide reason/explanation for this update.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] Check your code additions will not fail linting checks:
  - [ ] `go fmt -s`
  - [ ] `go vet`

### CODE REVIEW

File: providers/saml.go [hunk 1]
```diff
@@ -135,9 +137,9 @@ func (s *SAMLProvider) initialiseSAMLMiddleware() {
 		SAMLLogger.Debugf("SP acs URL: %v", acsURL.String())
 
 		var forceAuthn = s.config.ForceAuthentication
-
 		sp := saml.ServiceProvider{
-			EntityID:          metadataURL.String(),
+			// if s.config.EntityId is empty, it will default to metadataUrl
+			EntityID:          s.config.EntityId,
```
Comment (buger) [inline providers/saml.go:142]: Where is thee code which sets default value to `metadataURL.String()`?

Comment (sredxny) [reply providers/saml.go:142]: @buger inside the library, there we can find:

```
type ServiceProvider struct {
	// Entity ID is optional - if not specified then MetadataURL will be used
	EntityID string

	.
        .
        .
}
```

Comment (buger) [reply providers/saml.go:142]: So this line we had previously did nothing?

Comment (sredxny) [reply providers/saml.go:142]: Yes, it wasn't needed.

**Review by tbuchaillot — APPROVED** [review]

**Pull request merged**